mlflow/system_metrics/system_metrics_monitor.py

```diff
@@ -61,7 +62,12 @@ def __init__(
         # Instantiate default monitors.
         self.monitors = [CPUMonitor(), DiskMonitor(), NetworkMonitor()]
         try:
-            gpu_monitor = GPUMonitor()
+            if torch.cuda.is_available() and not torch.version.hip:
```

Comment:
Could it be that this requires an `import torch`?

Reply:
@mpvginde you're of course right, and this is obviously not desired. I updated the code to follow the same logic as before: a try/except block checking whether pynvml/pyrocml is installed without an if-statement depending on torch. See latest commit